### ISSUE TITLE
EMD is large and remains unchanged between epoches

### ISSUE THREAD

**Issue by Leong1230** [daerduoCarey/PyTorchEMD#10]
Hi! Thank you for your great work!
I have some questions and need your help.

1. I got a large EMD (from 100~1000) when training the model, and it didn't change throughout training (with changes smaller than 1). I divided the EMD loss by the number of points (1200~1600, with centerilze but without normalize). I wonder if I need to normalize the point coordinates between 0~1 before send it into EMD?
2. In tese_emd_loss.py, loss is computed by 'loss = d[0] / 2 + d[1] * 2 + d[2] / 3'. I wonder why the coefficients are different among different dims of d ?

Could you please give some advice?
Thanks!

**Comment by daerduoCarey** [comment]
1. you don't have to normalize point coordinates. you can normalize by point count if you want. EMD computes a sum, not an avg.
2. it's just a random test.

**Comment by Leong1230** [comment]
Thanks for your reply!

**Comment by Leong1230** [comment]
Hi! I found myself still trapped in this question, after I have tuned the hyperparamters and rechecked the model. 
My model takes an image as input and outputs a point cloud. The input images are normalized.
When I did normlization to the ground truth points, the training loss converges to a very small value (like 0.000198 in epoch 350/400). But it then exploded and finally converges to a larger value (like 0.328). (I have set 'shuffle=True' in the Dataloader)
When I didn't do normlization to gt, the training loss converges to a very large value (like around 794) even though I have tried different learning rates.
And interestingly, the norm of each point in my dataset in around 47, and the difference between 0.328 and 794 is around 47^2 times. But the model without a normalized gt didn't have a convergence process like that with a normalized gt.
And the model converges with Chamfer distance loss no matter that the gt is normalized or not.

Since I am new in this area and not familiar with the cuda implementation of EMD,  I have some stupid questions: 
1. In my application background, the shape size of the network output is important, which will be lost if I do the normalization. So could you please give me some advice to fix this?
2. Can I say it would be better to normalize the gt to train the model with this implementation of EMD?

Thanks!
